src/schema-generation/flex-search-filter-input-types/filter-fields.ts

```diff
@@ -0,0 +1,232 @@
+}
+
+export class FlexSearchScalarOrEnumFieldFilterField implements FlexSearchFilterField {
+    public readonly inputType: GraphQLInputType;
+    public readonly description: string | undefined;
+
+    constructor(
+        public readonly field: Field,
+        public readonly resolveOperator: (fieldNode: QueryNode, valueNode: QueryNode, flexSearchLanguage?: FlexSearchLanguage) => QueryNode,
+        public readonly operatorPrefix: string | undefined,
+        baseInputType: GraphQLInputType,
+        public readonly flexSearchLanguage?: FlexSearchLanguage
+    ) {
+        this.inputType = FLEX_SEARCH_OPERATORS_WITH_LIST_OPERAND.includes(operatorPrefix || '') ? new GraphQLList(new GraphQLNonNull(baseInputType)) : baseInputType;
+        this.description = getDescription({ operator: operatorPrefix, fieldName: field.name, typeName: field.type.name, isAggregation: this.operatorPrefix != undefined });
+
+        if (this.field.description) {
+            this.description = (this.description ? this.description + '\n\n' : '') + this.field.description;
+        }
+    }
+
+    get name() {
+        if (this.operatorPrefix == undefined) {
+            return this.field.name;
+        }
+        return this.field.name + FILTER_FIELD_PREFIX_SEPARATOR + this.operatorPrefix;
+    }
+
+    getFilterNode(sourceNode: QueryNode, filterValue: AnyValue, path: ReadonlyArray<Field>, info: QueryNodeResolveInfo): QueryNode {
+        const valueNode = new FieldPathQueryNode(sourceNode, path.concat(this.field));
+        const literalNode = new LiteralQueryNode(filterValue);
+        if ((this.operatorPrefix == undefined || this.operatorPrefix === '') && filterValue == null) {
+            return new BinaryOperationQueryNode(
+                this.resolveOperator(valueNode, literalNode, this.flexSearchLanguage),
+                BinaryOperator.OR,
+                not(new FlexSearchFieldExistsQueryNode(valueNode, this.flexSearchLanguage))
+            );
+        }
+        if ((this.operatorPrefix == INPUT_FIELD_NOT) && filterValue == null) {
+            return new BinaryOperationQueryNode(
+                this.resolveOperator(valueNode, literalNode, this.flexSearchLanguage),
+                BinaryOperator.AND,
+                new FlexSearchFieldExistsQueryNode(valueNode, this.flexSearchLanguage)
+            );
+        }
+
+        if (this.operatorPrefix == INPUT_FIELD_NOT_STARTS_WITH && (filterValue === '')) {
+            return new ConstBoolQueryNode(false);
+        }
+        if ((this.operatorPrefix == INPUT_FIELD_STARTS_WITH || this.operatorPrefix == INPUT_FIELD_NOT_STARTS_WITH || STRING_TEXT_ANALYZER_FILTER_FIELDS.some(value => this.operatorPrefix === value)) && (filterValue == null || filterValue === '')) {
+            return new ConstBoolQueryNode(true);
+        }
+
+
+        return this.resolveOperator(valueNode, literalNode, this.flexSearchLanguage);
```

Comment:
generally, avoid unnecessary newlines - empty lines between statements for grouping are good, but two empty lines or empty lines after the last statement are distracting IMO